flink-table/flink-table-api-java/pom.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-table</artifactId>
+		<version>1.8-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-table-api-java_${scala.binary.version}</artifactId>
+	<name>flink-table-api-java</name>
+
+	<packaging>jar</packaging>
+
+	<dependencies>
+		<!-- Base Table API classes. -->
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-table-api-base</artifactId>
+			<version>${project.version}</version>
+		</dependency>
+		<!-- Integration with Java DataSet/DataStream API. -->
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-streaming-java_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+			<scope>provided</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
```

Comment:
I'm probably just showing my ignorance here, but why do we need shading here? The DataStream API packages don't have shading.

Reply:
The reason was to include `flink-table-common` in the API modules. This allows to copy api and planner into the `/lib` folder and table programs should be executable.

Reply:
I think we should have the least amount of magic possible here. We might decide to have something like a `flink-table-dist` that bundles everything necessary for SQL programs, then `flink-table-common`, `flink-table-api`, `flink-table-planner` et al wouldn't even have to be in opt, only this fat table thing. WDYT?

Reply:
Yes, we could provide such a module. But would you also add `flink-table-api-java` and `flink-table-api-scala` there? If yes, they might overlap each other until we introduce an API breaking change for removing `TableEnvironment.getTableEnvironment(...)`. See the discussion in FLINK-11067. 
What do you mean with "wouldn't even have to be in opt"?